h2/stream.py

```diff
@@ -1049,9 +1050,12 @@ def _build_hdr_validation_flags(self, events):
             # an internal event, so we can do away with this branch.
             is_trailer = False
 
+        is_response_header = is_response and not is_trailer
```

Comment:
`is_response` should be completely determinable by the events. Do we really need the extra parameter?